### ISSUE TITLE
allow ibeacon presence tracking with any combination of uuid/major/minor

### ISSUE THREAD

**Issue by travisghansen** [esphome/feature-requests#806]
**Describe the problem you have/What new integration you would like**

I would like to use `ble_presence` with more than just `uuid` detection, but rather any combination of `uuid`, `major` and `minor`.

**Please describe your use case for this integration and alternatives you've tried:**

Currently I have an rpi setup with `Node-RED` spewing all ibeacon traffic over mqtt to a *separate* `Node-RED` instance which then has timer logic to detect the presence (or lack thereof) of any combination of uuid/major/minor. I'd like to migrate this away from the setup to esphome. However, the current `ble_presence`  (at least according to this: https://github.com/esphome/esphome/pull/800/files#diff-a3d0ea97c90b0587466ef56a0a4ac296 doc appears to be missing otherwise) only supports listening for `uuid`. I use the same `uuid` but tweak the `major` and `minor` values to my liking per 'thing' (in my case, ibeacons in various vehicles).

So I could for example create a unique `uuid`/`major` `ble_sensor` to detect *any* vehicle is present (all vehicles sharing those 2 values). However, I could also create multiple `ble_sensor`s with all 3 defined (only variance would be the `minor`) which would detect specific vehicles. 

**Additional context**

This is for example how the location tracking works in conjunction with HASS and the ios app. For example:

```
# look only for uuid
zone.home:
  beacon:
    uuid: 2F234454-CF6D-4A0F-ADF2-
    #major: 10
    #minor: 0

# must match uuid and major
zone.home:
  beacon:
    uuid: 2F234454-CF6D-4A0F-ADF2-
    major: 10
    #minor: 0

# etc, I believe you can even remove uuid altogether and just check for major or minor as well
# basically you 3 datapoints to test and only those defined are checked for relevance
```

Thanks for consideration and help!

**Comment by travisghansen** [comment]
- https://github.com/esphome/esphome/pull/800
- https://github.com/esphome/feature-requests/issues/94
